clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java

```diff
@@ -102,21 +104,21 @@ public void testValuesWithPrefixOverride() {
         Map<String, Object> valuesWithPrefixOverride = config.valuesWithPrefixOverride(prefix);
 
         // prefix overrides global
-        assertTrue(config.unused().contains("prefix.sasl.mechanism"));
+        assertTrue(config.unknown().contains("prefix.sasl.mechanism"));
         assertTrue(config.unused().contains("sasl.mechanism"));
         assertEquals("GSSAPI", valuesWithPrefixOverride.get("sasl.mechanism"));
         assertFalse(config.unused().contains("sasl.mechanism"));
         assertFalse(config.unused().contains("prefix.sasl.mechanism"));
 
         // prefix overrides default
-        assertTrue(config.unused().contains("prefix.sasl.kerberos.kinit.cmd"));
+        assertTrue(config.unknown().contains("prefix.sasl.kerberos.kinit.cmd"));
         assertFalse(config.unused().contains("sasl.kerberos.kinit.cmd"));
         assertEquals("/usr/bin/kinit2", valuesWithPrefixOverride.get("sasl.kerberos.kinit.cmd"));
         assertFalse(config.unused().contains("sasl.kerberos.kinit.cmd"));
         assertFalse(config.unused().contains("prefix.sasl.kerberos.kinit.cmd"));
```

Comment:
same here and below, we should assert false to `(config.unknown().contains("prefix.sasl.kerberos.kinit.cmd")` after using that config.